### PR TITLE
fix: Windows and macOS installs

### LINKED ISSUE #13: Installation from npm fails
```
$ npm install -g go-ipfs
npm ERR! path /Users/danny/.nvm/versions/node/v8.8.1/lib/node_modules/go-ipfs/bin/ipfs
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod
npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/danny/.nvm/versions/node/v8.8.1/lib/node_modules/go-ipfs/bin/ipfs'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

$ npm -v
5.5.1

$ node -v
v8.8.1
```

### PR DESCRIPTION
**Pull request by hacdias**
Fixes #13.

### CODE REVIEW

**Comment by hacdias** [comment]
@VictorBjelkholm Circle is failing here since it is using node 0.10 which I think it's kind of very ooooooold. Could you sync the settings from [ci-sync](https://github.com/ipfs/ci-sync) with this repo?

**Comment by hacdias** [comment]
Sure, what should I do to move this to Jenkins?

File: link-ipfs.js [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
 
-var fs = require('fs')
-var path = require('path')
-var cproc = require('child_process')
+const fs = require('fs')
+const path = require('path')
+const cproc = require('child_process')
+const isWin = process.platform === 'win32'
 var version = require('./package.json').version
 version = version.replace(/-hacky[0-9]+/, '') // remove superfluous -suffix.
```
Comment (dryajov) [inline link-ipfs.js]: Why is this needed?

File: link-ipfs.js [hunk 2]
```diff
@@ -1,23 +1,37 @@
...
+  depBin += '.exe'
+  localBin += '.exe'
+}
 
 if (!fs.existsSync(depBin)) {
   die('ipfs binary not found. maybe go-ipfs-dep did not install correctly?')
 }
 
-fs.unlinkSync(localBin)
+if (fs.existsSync(localBin)) fs.unlinkSync(localBin)
```
Comment (dryajov) [inline link-ipfs.js]: I don't think this will lint properly with standard style?

File: link-ipfs.js (hunk 1 above)
Comment (hacdias) [reply link-ipfs.js]: I just changed some `vars` to `const`.

**Comment by hacdias** [comment]
@dryajov just tweaked some stuff.

File: link-ipfs.js (hunk 1 above)
Comment (hacdias) [reply link-ipfs.js]: If you meant the version variable, it is used down the script.

**Comment by hacdias** [comment]
@VictorBjelkholm could you also remove Circle and Travis please? (It says 'forbidden' when I try).

File: link-ipfs.js [hunk 3]
```diff
@@ -4,8 +4,8 @@ const fs = require('fs')
 const path = require('path')
 const cproc = require('child_process')
 const isWin = process.platform === 'win32'
-var version = require('./package.json').version
-version = version.replace(/-hacky[0-9]+/, '') // remove superfluous -suffix.
+const version = require('./package.json').version
+  .replace(/-hacky[0-9]+/, '') // remove superfluous -suffix.
```
Comment (dryajov) [inline link-ipfs.js]: I'm still not sure we need the replace... Do we still have the `-hacky` convention? 

// cc @VictorBjelkholm @diasdavid

**Comment by VictorBjelkholm** [comment]
@hacdias Travis has now been disabled. Where did you try on CircleCI? AFAIK, to disable a project, you need to get everyone who follows it to stop following it, otherwise it'll continue building.

**Comment by hacdias** [comment]
@VictorBjelkholm if you follow the project, then there will be an option called 'Stop Building' which automatically turns the project to 0 unfollowers

image

**Comment by hacdias** [comment]
@dryajov here's the reason for `hacky`: https://github.com/ipfs/npm-go-ipfs#publish-a-new-version-of-this-module-with-exact-same-go-ipfs-version

**Comment by dryajov** [comment]
Thanks @hacdias, that makes sense now 👍

**Comment by hacdias** [comment]
@diasdavid could you release a new version of this, please?

**Pull request merged**